src/plugins/data/public/antlr/promql/code_completion.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { monaco } from '@osd/monaco';
+import { CursorPosition, PromQLAutocompleteResult } from '../shared/types';
+import { openSearchPromQLAutocompleteData } from './promql_autocomplete';
+import { QuerySuggestion, QuerySuggestionGetFnArgs } from '../../autocomplete';
+import { parseQuery } from '../shared/utils';
+import { SuggestionItemDetailsTags } from '../shared/constants';
+
+export interface SuggestionParams {
+  position: monaco.Position;
+  query: string;
+}
+
+export interface ISuggestionItem {
+  text: string;
+  type: string;
+  fieldType?: string;
+}
+
+export const getSuggestions = async ({
+  selectionStart,
+  selectionEnd,
+  indexPattern,
+  dataset,
+  position,
+  query,
+  services,
+}: QuerySuggestionGetFnArgs): Promise<QuerySuggestion[]> => {
+  if (!services || !services.appName || !indexPattern || !dataset) return [];
+  try {
+    const { lineNumber, column } = position || {};
+    const suggestions = getOpenSearchPromQLAutoCompleteSuggestions(query, {
+      line: lineNumber || selectionStart,
+      column: column || selectionEnd,
+    });
+    const prometheusResourceClient = services.data.resourceClientFactory.create('prometheus');
```

Comment:
I'd suggest passing an instance of the `PrometheusResourceClient` into `getSuggetions`. That way we can re-use the instance across multiple suggestions.

Reply:
Thanks for the catch! I'll make this change